tests/rendering.rs

```diff
@@ -241,6 +241,7 @@ fn check(state: &LayoutState, expected_hash_data: &str, name: &str) {
     check_dims(state, [300, 500], expected_hash_data, name);
 }
 
+#[track_caller]
 fn check_dims(state: &LayoutState, dims: [usize; 2], expected_hash_data: &str, name: &str) {
```

Comment:
This is a cool change. If the check fails, the error is marked where the caller called this function.